explorer/src/providers/accounts/rewards.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import { Cluster, useCluster } from "providers/cluster";
+import * as Cache from "providers/cache";
+import { Connection, InflationReward, PublicKey } from "@solana/web3.js";
+import { ActionType } from "providers/block";
+import { FetchStatus } from "providers/cache";
+import { reportError } from "utils/sentry";
+
+const PAGE_SIZE = 15;
+
+export type Rewards = {
+  highestFetchedEpoch?: number;
+  lowestFetchedEpoch?: number;
+  rewards: (InflationReward | null)[];
+  foundOldest?: boolean;
+};
+
+export type RewardsUpdate = {
+  rewards: (InflationReward | null)[];
+  foundOldest?: boolean;
+};
+
+type State = Cache.State<Rewards>;
+type Dispatch = Cache.Dispatch<RewardsUpdate>;
+
+function reconcile(
+  rewards: Rewards | undefined,
+  update: RewardsUpdate | undefined
+): Rewards | undefined {
+  if (update?.rewards === undefined) {
+    return rewards;
+  }
+
+  const combined = (rewards?.rewards || [])
+    .concat(update.rewards)
+    .filter((value) => value !== null);
+
+  const foundOldest =
+    update.foundOldest ||
+    update.rewards.length < 1 ||
+    update.rewards[update.rewards.length - 1]?.epoch ===
+      combined[combined.length]?.epoch;
```

Comment:
I don't really understand this condition. Does this mean that after we fetch once, we won't fetch anymore?